tests/fixtures/logs.py

```diff
@@ -24,3 +29,101 @@ def gelf_logger():
+        "Mozilla/5.0 (Windows NT 10.0; Win64; x64; XBOX_ONE_ED) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393",
+        "Mozilla/5.0 (Windows Phone 10.0; Android 4.2.1; Xbox; Xbox One) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2486.0 Mobile Safari/537.36 Edge/13.10586",
+        "Mozilla/5.0 (PlayStation 4 3.11) AppleWebKit/537.73 (KHTML, like Gecko)",
+        "Mozilla/5.0 (PlayStation Vita 3.61) AppleWebKit/537.73 (KHTML, like Gecko) Silk/3.2",
+        "Mozilla/5.0 (Nintendo 3DS; U; ; en) Version/1.7412.EU",
+        "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)",
+        "Mozilla/5.0 (compatible; bingbot/2.0; +http://www.bing.com/bingbot.htm)",
+        "Mozilla/5.0 (compatible; Yahoo! Slurp; http://help.yahoo.com/help/us/ysearch/slurp)",
+        "Mozilla/5.0 (X11; U; Linux armv7l like Android; en-us) AppleWebKit/531.2+ (KHTML, like Gecko) Version/5.0 Safari/533.2+ Kindle/3.0+",
+        "Mozilla/5.0 (Linux; U; en-US) AppleWebKit/528.5+ (KHTML, like Gecko, Safari/528.5+) Version/4.0 Kindle/3.0 (screen 600x800; rotate)"
+    ]
+
+    return random.choice(agents)
+
+
+def random_string_generator(size, chars=string.printable):
+    """returns a random string of size 'size' including characters in 'chars' """
+    return ''.join(random.choice(chars) for _ in range(size))
+
+@pytest.fixture
+def log_generator():
+    """Generate one common json event and the expected parsing output"""
+    data = {}
+    data['username'] = random_string_generator(random.randrange(0, 300))
+    data['ip'] = ".".join(str(random.randint(0, 255)) for _ in range(4))
+    data['agent'] = random_user_agent_generator()
+    data['accept_language'] = 'en-US;q=0.5,en;q=0.3' #todo - more variants of this header
+    data['event_source'] = 'server'
+    data['time'] = (str(random.randint(2006, 2050))
+        + "-" + str(random.randint(1, 12))
+        + "-" + str(random.randint(1, 31))
+        + "T" + str(random.randint(0, 24))
+        + ":" + str(random.randint(1, 60)) #todo - not really correct as should be 01 instead of 1
+        + ":" + str(random.randint(1, 60))
+        + ":" + str(random.randint(1, 1000000)) + "00:00")
+    data['host'] = random_string_generator(12)
+    data['referer'] = random_string_generator(random.randint(1, 1000), URI_ALLOWED_CHARS)
+    data['event_type'] = random_string_generator(random.randint(1, 1000), URI_ALLOWED_CHARS)
+    raw_json_event = json.dumps(data)
+
+    expected_parsed_event = pd.DataFrame({
+        "username": [data['username']],
+        "ip": [data['ip']],
+        "agent": [data['agent']],
+        "accept_language": [data['accept_language']],
+        "event_source": data['event_source'],
+        "time": [data['time']],
+        "host": [data['host']],
+        "referer": [data['referer']],
+        "event_type": [data['event_type']],
+    })
+    return raw_json_event, expected_parsed_event
```

Comment:
IMO we don't need to JSON intermediary, I would suggest that you only return the factory, _e.g._:

```suggestion
    def _event(n):
      return pd.DataFrame(data=[{
          "username": fake.profile().get("username"),
          # [...]
      } for i in range(n)])
    return _event
```

For reference, see: https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures

Reply:
Thanks, I removed this part. Initially I thought the factory could provide the expected output too, but you're right - this tests more the pandas JSON conversion :)